hibernate-core/src/test/java/org/hibernate/orm/test/procedure/SybaseStoredProcedureTest.java

```diff
@@ -119,15 +127,18 @@ public void testStoredProcedureOutParameter(EntityManagerFactoryScope scope) {
 	@Test
 	public void testStoredProcedureOutParameterDifferentParametersRegistrationOrder(EntityManagerFactoryScope scope) {
 		scope.inTransaction( entityManager -> {
-			StoredProcedureQuery query = entityManager.createStoredProcedureQuery( "sp_count_phones" );
-			query.registerStoredProcedureParameter( "phoneCount", Long.class, ParameterMode.OUT );
-			query.registerStoredProcedureParameter( "personId", Long.class, ParameterMode.IN );
-
-			query.setParameter( "personId", 1L );
-
-			query.execute();
-			Long phoneCount = (Long) query.getOutputParameterValue( "phoneCount" );
-			assertEquals( Long.valueOf( 2 ), phoneCount );
+			// if named parameter are not supported then the parameter should be registered in the same order as defined in the stored procedure
+			if ( supportsNamedParameters( scope ) ) {
```

Comment:
Would be nice to add a `DialectFeatureChecks` for this and annotate `@RequiresDialectFeature`